src/pages/index.js

```diff
@@ -3,16 +3,139 @@ import FormValidator from "../components/FormValidator.js";
 import "./index.css";
 import PopupWithForm from "../components/PopupWithForm.js";
 import PopupWithImage from "../components/PopupWithImage.js";
+import confirmPopup from "../components/confirmPopup.js";
+import Popup from "../components/Popup.js";
 import Section from "../components/Section";
 import UserInfo from "../components/UserInfo.js";
-import { data, variables, formConfig } from "../utils/Constants.js";
+import { variables, formConfig } from "../utils/Constants.js";
+import Api from "../components/Api.js";
 
 //------------------------------------------------ Refactoring Code
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    Authorization: "44ea7345-0378-4698-8841-25c6e07018e7",
+    "Content-Type": "application/json",
+  },
+});
+
+const user = new UserInfo({
+  nameSelector: "#profile-info-title",
+  jobSelector: "#profile-info-description",
+  avatarSelector: "#user-avatar",
+});
+
+api
+  .getUserInfo()
+  .then((userData) => {
+    user.setUserInfo({
+      name: userData.name,
+      job: userData.about,
+      avatar: userData.avatar,
+    });
+    console.log("User information loaded and displayed successfully.");
+  })
+  .catch((error) => {
+    console.error("Failed to load user information:", error);
+  });
+
+const editFormPopup = new PopupWithForm("#profile-edit-modal", (formData) => {
+  variables.editModalProfileSaveButton.textContent = "Saving...";
```

Comment:
**Could be improved**

It’s better to make method `renderLoading` inside class `PopupWithForm`. You should find the button once in the constructor and change the text:  

```js
  constructor({ popupSelector, handleSubmit }) {
    super(popupSelector);
    // fix the initial button text only once in the constructor
    this._submitBtnText = this._submitBtn.textContent
  }
  // add 2 params: isLoading and loadingText with a default text
  renderLoading(isLoading, loadingText='Saving...') {
    if (isLoading) {
      this._submitBtn.textContent = loadingText;
    } else {
	// here we return back the initial text. So, you don’t need to bother yourself about it
      this._submitBtn.textContent = this._submitBtnText;
    }
  }

```